### ISSUE TITLE
DeliveryReceipt: A member with the name 'status' already exists on 'Nexmo.Api.Messaging.DeliveryReceipt'. Use the JsonPropertyAttribute to specify another name.'

### ISSUE THREAD

**Issue by surexxx** [Vonage/vonage-dotnet-sdk#212]
I have updated the Nexmo.Csharp.Client nuget from version 4.0.0 to 5.2.0. Updated the code sending sms and receiving dlr report using code snippets from here:
- https://developer.nexmo.com/messaging/sms/code-snippets/send-an-sms
- https://developer.nexmo.com/messaging/sms/code-snippets/delivery-receipts

When testing receiving delivery report from postman with this payload (from here: https://developer.nexmo.com/messaging/sms/code-snippets/delivery-receipts):
{
  "err-code": "0",
  "message-timestamp": "2020-10-25 12:10:29",
  "messageId": "0B00000127FDBC63",
  "msisdn": "447700900000",
  "network-code": "23410",
  "price": "0.03330000",
  "scts": "1810251310",
  "status": "delivered",
  "to": "Nexmo CLI"
}
...I'am getting this error:
**DeliveryReceipt: A member with the name 'status' already exists on 'Nexmo.Api.Messaging.DeliveryReceipt'. Use the JsonPropertyAttribute to specify another name.'**

The .net project uses dotnetcore 2.2.

I think something is wrong in the new library around DeliveryReceipt class, or documentation is not up to date.

Is there any workaround or solution for this issue?

**Comment by slorello89** [comment]
Hi @surexxx thanks for bringing this to my attention:

I notice that the `Status` attribute isn't being ignored in the [DeliveryReceipt](https://github.com/Nexmo/nexmo-dotnet/blob/master/Nexmo.Api/Messaging/DeliveryReceipt.cs#L42) class so I'm wondering if that's where this is coming from (that's an enum being parsed from the string status). However I do have a [unit test](https://github.com/Nexmo/nexmo-dotnet/blob/master/Nexmo.Api.Test.Unit/MessagingTests.cs#L196) testing essentially this exact code path, which is passing. Can you share which version of Newtonsoft you're using? Maybe that's where the disconnect is

**Comment by surexxx** [comment]
Hi @slorello89 

Thanks for rapid answer. I'm using Newtonsoft.Json 11.0.2.

**Comment by slorello89** [comment]
@surexxx - Hmm Newtonsoft doesn't seem to be the problem, I still can't reproduce it using Newtonsoft 11.0.2/core2.2. Do you have a sample project where this is happening I could possibly look at?

**Comment by surexxx** [comment]
@slorello89 - I'll try to make a sample project.

**Comment by surexxx** [comment]
@slorello89  - I think I found the problem. In startup.cs I used a default contract serializer:
`JsonConvert.DefaultSettings = () =>
                new JsonSerializerSettings
                {
                    ContractResolver = new CamelCasePropertyNamesContractResolver()
                };`
After these lines removed, it's working. Try it pls.

**Comment by slorello89** [comment]
Yep that's definitely it, I opened #214 to address this. In the interim you can work around this telling the serializer to use the `DefaultContractResolver` when you are deserializing the DLR

```csharp
 DeliveryReceipt dlr;
    using (StreamReader reader = new StreamReader(Request.Body, Encoding.UTF8))
    {
        var settings = new JsonSerializerSettings { ContractResolver = new DefaultContractResolver() }; // create the new settings
        dlr = JsonConvert.DeserializeObject<DeliveryReceipt>(reader.ReadToEndAsync().Result, settings); // use the new settings in the deserialization
    }
    Console.WriteLine($"Delivery receipt received for messages {dlr.MessageId}");
    return NoContent();
```

**Comment by surexxx** [comment]
Thanks a lot @slorello89 👍

**Comment by Cereal-Killa** [comment]
> Hi @surexxx thanks for bringing this to my attention:
> 
> I notice that the `Status` attribute isn't being ignored in the [DeliveryReceipt](https://github.com/Nexmo/nexmo-dotnet/blob/master/Nexmo.Api/Messaging/DeliveryReceipt.cs#L42) class so I'm wondering if that's where this is coming from (that's an enum being parsed from the string status). However I do have a [unit test](https://github.com/Nexmo/nexmo-dotnet/blob/master/Nexmo.Api.Test.Unit/MessagingTests.cs#L196) testing essentially this exact code path, which is passing. Can you share which version of Newtonsoft you're using? Maybe that's where the disconnect is (if that's the problem then I will add the Ignore to the Status - I just want to be able to reproduce this bug)

Since Status has been renamed to StringStatus this issue doesn't' apply anymore on the latest revision right?

**Comment by surexxx** [comment]
@Cereal-Killa the version I'm using is Newtonsoft.Json Version=11.0.0.0

**Comment by slorello89** [comment]
This item has been resolved as of 5.6.3.
